benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +44,196 @@ def get_data_model_from_task(task, param_dict):
+    has a witness option or not.
+
+    @param options: List of existing options
+    @param task: An instance of a task
+    @param witness_options: List of options which need to be set to handle witnesses
+        e.g. if the options are ["-w"], then the witness should be passed as "-w witness_file"
+        to the tool
+
+    @return: List of additional options to handle witnesses. They are constructed
+        based on witness_options. For example if witness_options is ["-w"], then
+        the return value will be ["-w", "witness_file"]
+    """
+
+    additional_options = []
+    if isinstance(task.options, dict) and "witness" in task.options.keys():
+        if not any(witness_option in options for witness_option in witness_options):
+            for witness_option in witness_options:
+                additional_options += [witness_option, get_witness(task)]
+        else:
+            raise benchexec.tools.template.UnsupportedFeatureException(
+                "You are passing a witness as both an option and through the task definition. "
+                "Please remove one of them."
+            )
+    return additional_options
+
+
+class TaskFilesConsidered(Enum):
+    """
+    Enum to represent the different types of input
+    files that can be considered for a task.
+    """
+
+    INPUT_FILES_OR_IDENTIFIER = 1
+    INPUT_FILES = 2
+    SINGLE_INPUT_FILE = 3
+
+
+def handle_witness_of_task(
+    task, options, witness_options, task_files_considered: TaskFilesConsidered
+):
+    """
+    This function returns the input files and witness options for a task.
+    The input files are based on the task_files_considered parameter.
+    The witness options are based on the witness_options parameter.
+
+    In validation task definition files, there is a key "witness" inside the
+    options of a task. To handle this type of tasks correctly in the tool info module
+    you should do the following two steps, if your tool does not already support
+    them natively:
+    1. Add the witness options obtained by calling the function `get_witness_options`
+        to the options list in the `cmdline` function.
+    2. Filter the input files to only get those tasks which are not witnesses. This
+        can be done by calling one of the following functions, depending on the
+        type of input files you want:
+            * `get_non_witness_input_files_or_identifier`: If you want to consider
+                both input files and identifiers
+            * `get_non_witness_input_files`: If you want to consider only input files
+            * `get_single_non_witness_input_file`: If you want to consider only a single
+                input file
+    This function takes care of the above two steps and returns the input files and
+    witness options.
+
+    @param task: An instance of a task
+    @param options: List of existing options
+    @param witness_options: List of options which need to be set to handle witnesses
```

Comment:
Didn't we discuss that this should not be a list? If it remains a list, the documentation at least needs to explain what happens if there are more than one entries.

Same for `get_witness_options`, of course.

In principle it would also be possible to support only one option in the high-level API and more possibility in the low-level API, if this is useful.

Reply:
Sorry I forgot to implement the change when we discussed it.

I like the compromise to have a high-level API `handle_witness_of_task` which takes only a single option and the low level API `get_witness_options` taking a list. Addressed in 1c4bcbdbc9b9c9a94fc2cf191ce5d1babed69090